lib/pomelo_ex/general/users/modify_user.ex

```diff
@@ -1,7 +1,34 @@
 defmodule PomeloEx.General.Users.ModifyUser do
   @moduledoc false
+  alias PomeloEx.General.Authorization
+  alias PomeloEx.Types.General.Users.ModifyUserType
 
-  def execute(payload) do
-    raise "Not implemented #{__MODULE__} payload:" <> inspect(payload)
+  def execute(%ModifyUserType{id: id} = payload) do
+    http_client = Application.get_env(:pomelo_ex, :http_adapter)
+    url = Application.get_env(:pomelo_ex, :url)
+    headers = build_headers()
+
+    body =
+      payload
+      |> Map.from_struct()
+      |> Enum.reject(fn {_, v} -> v == nil end)
+      |> Enum.into(%{})
+      |> Jason.encode!()
+
+    http_client.patch("#{url}/users/v1/#{id}", body, headers)
+  end
+
+  defp build_headers do
+    {:ok, %HTTPoison.Response{status_code: 200, body: body}} = Authorization.request_token()
```

Comment:
I would need to pass this token as a parameter to the execute function. The idea behind this token is for the client to reuse it until it expires; however, in this case, a new token is being generated with each request.